sky/backends/backend_utils.py

```diff
@@ -1839,14 +1855,87 @@ def refresh_cluster_status_handle(
     if isinstance(handle, backends.CloudVmRayBackend.ResourceHandle):
         use_spot = handle.launched_resources.use_spot
         if (force_refresh or record['autostop'] >= 0 or use_spot):
-            record = _update_cluster_identity_and_status(
-                cluster_name,
-                acquire_per_cluster_status_lock=acquire_per_cluster_status_lock)
-            if record is None:
-                return None, None
+            try:
+                record = _update_cluster_identity_and_status(
+                    cluster_name,
+                    acquire_per_cluster_status_lock=
+                    acquire_per_cluster_status_lock)
+                if record is None:
+                    return None, None
+            except (exceptions.ClusterOwnerIdentityMismatchError,
+                    exceptions.CloudUserIdentityError,
+                    exceptions.ClusterStatusFetchingError) as e:
+                if suppress_error:
+                    logger.debug(
+                        f'Failed to refresh cluster {cluster_name!r} due to {e}'
+                    )
+                    return None, None
+                raise
     return record['status'], record['handle']
 
 
+def check_cluster_available(
+    cluster_name: str,
+    operation: str,
+    *,
+    expected_backend: Optional[Type[backends.Backend]] = None
```

Comment:
Done for 1.

For 2, since the `execution.exec` uses this function without a expected backend, we need to have way to turn off the check for hte backend. I changed it to a bool to make it more specific and most of the callers will not need to pass the argument. Wdyt?